spec/lib/pedicel/validator/token_data_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'pedicel/validator'
+require 'lib/pedicel/validator/helper'
+
+describe Pedicel::Validator::TokenData do
+  let(:token_data_h) { JSON.parse(token.unencrypted_data.to_hash.to_json) }
+  let(:validator) { Pedicel::Validator::TokenData.new(token_data_h.to_hash) }
+
+  describe '#valid?' do
+    it 'relies on #validate' do
+      return_value = 'return_value'
+      expect(validator).to receive(:validate).and_return(return_value)
+      expect(validator.valid?).to equal return_value
+    end
+  end
+
+  describe '#validate' do
+    context 'can be happy' do
+      subject { lambda { validator.validate } }
+
+      it 'does not err on a valid token' do
+        is_expected.to_not raise_error
+      end
+
+      it 'is truthy for a valid token' do
+        expect(validator.validate).to be_truthy
```

Comment:
Probably change to this:
```diff
       it 'is truthy for a valid token' do
-        expect(validator.validate).to be_truthy
+        is_expected.to be_truthy
       end
```

Reply:
Nope, because `subject` is a lambda, thus, `is_expected.to be_truthy` would test that the lambda is truthy which it always is.

Reply:
However, the lambda does not always, when `.call`'ed evaluate to something truthy.

Alternatively, we could use
```ruby
expect(subject.call).to be_truthy
```

I guess that'd be better?

Reply:
Even better:

```diff
diff --git a/spec/lib/pedicel/validator/token_data_spec.rb b/spec/lib/pedicel/validator/token_data_spec.rb
index b77c998..beda440 100644
--- a/spec/lib/pedicel/validator/token_data_spec.rb
+++ b/spec/lib/pedicel/validator/token_data_spec.rb
@@ -15,14 +15,12 @@ describe Pedicel::Validator::TokenData do

   describe '#validate' do
     context 'can be happy' do
-      subject { lambda { validator.validate } }
-
       it 'does not err on a valid token' do
-        is_expected.to_not raise_error
+        expect{validator.validate}.to_not raise_error
       end

       it 'is truthy for a valid token' do
-        expect(validator.validate).to be_truthy
+        expect(validator.validate).to be_truthyr
       end
     end
   end
```

Reply:
:+1: Très bien